pkcs7.go

```diff
@@ -213,6 +215,24 @@ func parseEncryptedData(data []byte) (*PKCS7, error) {
 	}, nil
 }
 
+// EnableFallbackLegacyX509CertificateParser parsing certificates
+// embedded in a PKCS7 message using the logic from crypto/x509 from before
+// Go 1.23. Go 1.23 introduced a breaking change in case a certificate contains
+// a critical authority key identifier, which is the correct thing to do based
+// on RFC 5280, but it breaks Windows devices performing the Simple Certificate
+// Enrolment Protocol (SCEP), as the certificates embedded in those requests
+// apparently have authority key identifier extensions marked critical.
+//
+// See https://go-review.googlesource.com/c/go/+/562341 for the change in the
+// Go source.
+//
+// When [EnableFallbackLegacyX509CertificateParser] is set to true, it'll first
+// try to parse the certificates using the regular Go crypto/x509 package, but
+// if it fails on the above case, it'll retry parsing the certificates using a
+// copy of the crypto/x509 package based on Go 1.23, but skips checking the
+// authority key identifier extension being critical or not.
+var EnableFallbackLegacyX509CertificateParser bool
```

Comment:
Perhaps a function that sets/unsets an atomic bit might be better here, instead of a global.

Reply:
b293aa74caf3dbbd8ddd79b01ac63937631326d6